### ISSUE TITLE
Output is different from expected

### ISSUE THREAD

**Issue by strogonoff** [ietf-ribose/doi2ietf-py#11]
I don’t know whether it’s a problem with the old Ruby library, or a consequence of porting to Python, but data returned from DOI is not converted entirely.

Can we make `doi2ietf` output a more complete structure, and use consistent schema with `relaton-data-` repositories (if that is the correct structure to expect)? This way BibXML service template would be able to display DOI references like others.

Details below.

## Request

If we request with curl an example DOI from crossref.org (the same URL that doi2ietf is using):

    curl -i -X GET -H "Accept: application/citeproc+json" https://data.crossref.org/DOI.10.6028/NIST.SP.800-53r5

## crossref.org response

We get this structure:

    {
        "institution": {
            "name": "National Institute of Standards and Technology",
            "place": [
                "Gaithersburg, MD"
            ],
            "department": [
                "Information Technology Laboratory"
            ],
            "acronym": [
                "NIST"
            ]
        },
        "indexed": {
            "date-parts": [
                [
                    2021,
                    9,
                    10
                ]
            ],
            "date-time": "2021-09-10T04:32:52Z",
            "timestamp": 1631248372363
        },
        "reference-count": 0,
        "publisher": "National Institute of Standards and Technology",
        "content-domain": {
            "domain": [],
            "crossmark-restriction": false
        },
        "DOI": "10.6028/nist.sp.800-53r5",
        "type": "report",
        "created": {
            "date-parts": [
                [
                    2020,
                    9,
                    23
                ]
            ],
            "date-time": "2020-09-23T18:26:43Z",
            "timestamp": 1600885603000
        },
        "source": "Crossref",
        "is-referenced-by-count": 4,
        "title": "Security and Privacy Controls for Information Systems and Organizations",
        "prefix": "10.6028",
        "author": [
            {
                "name": "Joint Task Force Interagency Working Group",
                "sequence": "first",
                "affiliation": []
            }
        ],
        "member": "4068",
        "published-online": {
            "date-parts": [
                [
                    2020,
                    9,
                    23
                ]
            ]
        },
        "container-title": [],
        "original-title": [],
        "deposited": {
            "date-parts": [
                [
                    2020,
                    9,
                    23
                ]
            ],
            "date-time": "2020-09-23T18:26:43Z",
            "timestamp": 1600885603000
        },
        "score": 1,
        "subtitle": [],
        "short-title": [],
        "issued": {
            "date-parts": [
                [
                    2020,
                    9,
                    23
                ]
            ]
        },
        "references-count": 0,
        "alternative-id": [
            "NIST SP 800-53r5"
        ],
        "URL": "http://dx.doi.org/10.6028/NIST.SP.800-53r5",
        "relation": {}
    }

## doi2ietf DICT result

If DICT output is requested, doi2ietf returns this structure:

    {
        "data": {
            "seriesinfo": {
                "DOI": "10.6028/nist.sp.800-53r5",
                "National Institute of Standards and Technology": "report"
            },
            "title": "Security and Privacy Controls for Information Systems and Organizations",
            "author": [
                {
                    "name": null
                }
            ],
            "date": "2020-09-23"
        }
    }

## doi2ietf XML result

If XML is requested, it crashes with AttributeError (#9). If I patch a file according like in #10, there is no crash but XML is also limited:

    <reference anchor="a">
        <front>
            <title>Security and Privacy Controls for Information Systems and Organizations</title>
            <author surname="">
                <organization></organization>
            </author>
            <date year="2020" month="September" day="23"/>
        </front>
        <seriesInfo name="DOI" value="10.6028/nist.sp.800-53r5"/>
        <seriesInfo name="National Institute of Standards and Technology" value="report"/>
    </reference>

**Comment by ronaldtse** [comment]
> Can we make doi2ietf output a more complete structure, and use consistent schema with relaton-data- repositories (if that is the correct structure to expect)? This way BibXML service template would be able to display DOI references like others.

Yes, this is the way to go. We may ignore the data insufficiency of the original Ruby implementation.

In fact we are supposed to have a `relaton-doi` implementation that handles the fetch from DOI. @andrew2net

**Comment by strogonoff** [comment]
Does it use crossref.org? Does it only handle fetch, or does it also handle conversion of DOI result to a Relaton structure? If it handles both, we could potentially make `doi2ietf` minimal and just proxy the call to `relaton-doi`, but I presume `relaton-doi` is not written in Python so it’s unclear how to invoke it.

**Comment by ronaldtse** [comment]
@strogonoff here we clearly need a Python mechanism to convert DOI to a Relaton structure. This probably should be added to the relaton-bib-py stack.
